pkg/controller/applicationintegration/sqs/contoller.go

```diff
@@ -0,0 +1,244 @@
+		return &external{client: queueClient, kube: c.kube}, errors.Wrap(err, errQueueClient)
+	}
+
+	if p.GetCredentialsSecretReference() == nil {
+		return nil, errors.New(errGetProviderSecret)
+	}
+
+	s := &corev1.Secret{}
+	n := types.NamespacedName{Namespace: p.Spec.CredentialsSecretRef.Namespace, Name: p.Spec.CredentialsSecretRef.Name}
+	if err := c.kube.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+
+	queueClient, err := c.newClientFn(ctx, s.Data[p.Spec.CredentialsSecretRef.Key], p.Spec.Region, awsclients.UseProviderSecret)
+	return &external{client: queueClient, kube: c.kube}, errors.Wrap(err, errQueueClient)
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.Queue)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotQueue)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	resAttributes, err := e.client.GetQueueAttributesRequest(&awssqs.GetQueueAttributesInput{
+		QueueUrl:       aws.String(meta.GetExternalName(cr)),
+		AttributeNames: []awssqs.QueueAttributeName{awssqs.QueueAttributeNameAll},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(sqs.IsErrorNotFound, err), errGetQueueAttributesFailed)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = v1alpha1.QueueObservation{
+		ARN: resAttributes.Attributes[v1alpha1.AttributeQueueArn],
+	}
+
+	resTags, err := e.client.ListQueueTagsRequest(&awssqs.ListQueueTagsInput{
+		QueueUrl: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errListQueueTagsFailed)
+	}
+
+	sqs.LateInitialize(&cr.Spec.ForProvider, resAttributes.Attributes, resTags.Tags)
+	current := cr.Spec.ForProvider.DeepCopy()
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: sqs.IsUpToDate(cr.Spec.ForProvider, resAttributes.Attributes, resTags.Tags),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Queue)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotQueue)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+
+	// FIFO queues names should start with ".fifo"
+	if aws.BoolValue(cr.Spec.ForProvider.FifoQueue) && !strings.HasSuffix(cr.Spec.ForProvider.Name, fifoQueueSuffix) {
+		return managed.ExternalCreation{}, errors.New(errInvalidNameForFifoQueue)
+	}
+
+	createResp, err := e.client.CreateQueueRequest(&awssqs.CreateQueueInput{
+		Attributes: sqs.GenerateCreateAttributes(&cr.Spec.ForProvider),
```

Comment:
Representing these attributes as a map is an unfortunate design choice in the AWS SDK :/ Thanks for making the most of this! :)